search.py

```diff
@@ -0,0 +1,46 @@
+import database_auth
+import requests
+import json
+from datetime import date
+from datetime import datetime
+from mysql.connector import Error
+
+connection = database_auth.connection()
+
+
+try:
+	if connection.is_connected():
+		cursor = connection.cursor()
+		fetch_query = 'select common_name from bikeapitest;'
```

Comment:
This is inefficient - you are transferring data for all bike points from the DB to your server in order to be able to make a query. Try to see if you can do the filtering of common name within the DB query itself so that you don't need to do so much data transfer